[2024-11-05T18:17:20.480+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-05T18:17:20.497+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_transaction_analysis.convert_to_btc scheduled__2024-11-04T00:00:00+00:00 [queued]>
[2024-11-05T18:17:20.505+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_transaction_analysis.convert_to_btc scheduled__2024-11-04T00:00:00+00:00 [queued]>
[2024-11-05T18:17:20.506+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-11-05T18:17:20.524+0000] {taskinstance.py:2888} INFO - Executing <Task(PythonOperator): convert_to_btc> on 2024-11-04 00:00:00+00:00
[2024-11-05T18:17:20.534+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=75) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-05T18:17:20.539+0000] {standard_task_runner.py:72} INFO - Started process 77 to run task
[2024-11-05T18:17:20.539+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'fraud_transaction_analysis', 'convert_to_btc', 'scheduled__2024-11-04T00:00:00+00:00', '--job-id', '87', '--raw', '--subdir', 'DAGS_FOLDER/fraud_transaktion_analysis-dag.py', '--cfg-path', '/tmp/tmphavrpbqp']
[2024-11-05T18:17:20.542+0000] {standard_task_runner.py:105} INFO - Job 87: Subtask convert_to_btc
[2024-11-05T18:17:20.612+0000] {task_command.py:467} INFO - Running <TaskInstance: fraud_transaction_analysis.convert_to_btc scheduled__2024-11-04T00:00:00+00:00 [running]> on host 21d07c241839
[2024-11-05T18:17:20.710+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='fraud_transaction_analysis' AIRFLOW_CTX_TASK_ID='convert_to_btc' AIRFLOW_CTX_EXECUTION_DATE='2024-11-04T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-04T00:00:00+00:00'
[2024-11-05T18:17:20.712+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-11-05T18:17:20.713+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-11-05T18:17:20.714+0000] {logging_mixin.py:190} INFO - Current task name:convert_to_btc state:running start_date:2024-11-05 18:17:20.497856+00:00
[2024-11-05T18:17:20.716+0000] {logging_mixin.py:190} INFO - Dag name:fraud_transaction_analysis and current dag run status:running
[2024-11-05T18:17:20.717+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-05T18:17:20.719+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/fraud_transaktion_analysis-dag.py", line 25, in convert_to_btc
    return total_amount_usd / bitcoin_rate
           ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
TypeError: unsupported operand type(s) for /: 'str' and 'str'
[2024-11-05T18:17:20.748+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-11-05T18:17:20.749+0000] {logging_mixin.py:190} INFO - Task start:2024-11-05 18:17:20.497856+00:00 end:2024-11-05 18:17:20.746929+00:00 duration:0.249073
[2024-11-05T18:17:20.750+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): convert_to_btc> dag:<DAG: fraud_transaction_analysis> dagrun:<DagRun fraud_transaction_analysis @ 2024-11-04 00:00:00+00:00: scheduled__2024-11-04T00:00:00+00:00, state:running, queued_at: 2024-11-05 18:17:05.191094+00:00. externally triggered: False>
[2024-11-05T18:17:20.751+0000] {logging_mixin.py:190} INFO - Failure caused by unsupported operand type(s) for /: 'str' and 'str'
[2024-11-05T18:17:20.754+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=fraud_transaction_analysis, task_id=convert_to_btc, run_id=scheduled__2024-11-04T00:00:00+00:00, execution_date=20241104T000000, start_date=20241105T181720, end_date=20241105T181720
[2024-11-05T18:17:20.783+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-05T18:17:20.788+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 87 for task convert_to_btc (unsupported operand type(s) for /: 'str' and 'str'; 77)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/fraud_transaktion_analysis-dag.py", line 25, in convert_to_btc
    return total_amount_usd / bitcoin_rate
           ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
TypeError: unsupported operand type(s) for /: 'str' and 'str'
[2024-11-05T18:17:20.838+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-05T18:17:20.867+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/models/baseoperator.py:1378: AirflowProviderDeprecationWarning: Call to deprecated class PostgresOperator. (Please use `***.providers.common.sql.operators.sql.SQLExecuteQueryOperator`.Also, you can provide `hook_params={'schema': <database>}`.)
  result = cls.__new__(cls)

[2024-11-05T18:17:20.912+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-05T18:17:20.932+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
